subsys/storage/flash_map/Kconfig

```diff
@@ -21,6 +21,19 @@ config FLASH_MAP_SHELL
 	help
 	  This enables shell commands to list and test flash maps.
 
+
+config FLASH_MAP_DEV_NAME_IN_FLASH_AREA
```

Comment:
Maybe for backward compatibility this option should impose device binding behavior in the code?
``` C
#ifdef CONFIG_FLASH_MAP_DEV_NAME_IN_FLASH_AREA
	if (fa->fa_dev == NULL) {
		flash_dev = device_get_binding(fa->fa_dev_name);
		if (flash_dev == NULL) {
			return -ENODEV;	
		}
	}

#endif
```

Reply:
I have such variant of the PR if desired. I have thought about the options and I now think, since that we allow custom flash_map to be developed, that user may fail to deliver device bindings so the option to use the new compile time resolution should be default for non custom maps.

Reply:
I agree with you.

Reply:
#ifdef CONFIG_FLASH_MAP_DEV_NAME_IN_FLASH_AREA ||
	if (fa->fa_dev == NULL) {
		flash_dev = device_get_binding(fa->fa_dev_name);
		if (flash_dev == NULL) {
			return -ENODEV;	
		}
	}
#endif
```